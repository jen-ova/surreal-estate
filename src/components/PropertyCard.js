import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faFortAwesome } from "@fortawesome/free-brands-svg-icons";
import { faBath, faBed } from "@fortawesome/free-solid-svg-icons";
import { faEnvelope } from "@fortawesome/free-regular-svg-icons";
import PropTypes from "prop-types";
import "../styles/propertyCard.css";

const PropertyCard = (props) => {
  const { title, type, bathrooms, bedrooms, price, city, email } = props;

  return (
    <div className="property-card__container">
      <div className="property-card__icon">
        <FontAwesomeIcon icon={faFortAwesome} />
      </div>
      <div className="property-card property-card__title">{title}</div>
      <div className="property-card property-card__typeCity">
        {type} - {city}
      </div>
      <div className="property-card">
        <FontAwesomeIcon icon={faBath} /> {bathrooms}
      </div>
      <div className="property-card">
        <FontAwesomeIcon icon={faBed} /> {bedrooms}
      </div>
      <div className="property-card property-card__price">Â£ {price}</div>
      <div className="property-card__email">
        <a href={`mailto:${email}`}>
          <FontAwesomeIcon icon={faEnvelope} /> Email
        </a>
      </div>
    </div>
  );
};

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  bathrooms: PropTypes.number.isRequired,
  bedrooms: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  city: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
};

export default PropertyCard;
