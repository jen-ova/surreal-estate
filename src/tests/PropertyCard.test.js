import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "Shinra Mansion",
    type: "Cottage",
    bathrooms: 4,
    bedrooms: 6,
    price: 2500,
    city: "Nibelheim",
    email: "sephiroth@shinra.com",
  };

  it("renders correctly", () => {
    const { asFragment } = render(<PropertyCard props={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  // it("renders correct values for props", () => {
  //   const { getByText } = render(<PropertyCard props={validProps} />);

  //   expect(getByText("Shinra Mansion")).toHaveClass("property-card__title");
  // });
});
