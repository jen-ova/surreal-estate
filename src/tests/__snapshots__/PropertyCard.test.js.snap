// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card__container"
  >
    <div
      class="property-card__icon"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-fort-awesome fa-w-16 "
        data-icon="fort-awesome"
        data-prefix="fab"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M489.2 287.9h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6V146.2c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6v-32c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6v-32c0-6-8-4.6-11.7-4.6v-38c8.3-2 17.1-3.4 25.7-3.4 10.9 0 20.9 4.3 31.4 4.3 4.6 0 27.7-1.1 27.7-8v-60c0-2.6-2-4.6-4.6-4.6-5.1 0-15.1 4.3-24 4.3-9.7 0-20.9-4.3-32.6-4.3-8 0-16 1.1-23.7 2.9v-4.9c5.4-2.6 9.1-8.3 9.1-14.3 0-20.7-31.4-20.8-31.4 0 0 6 3.7 11.7 9.1 14.3v111.7c-3.7 0-11.7-1.4-11.7 4.6v32h-36.6v-32c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32H128v-32c0-2.6-2-4.6-4.6-4.6H96c-2.6 0-4.6 2-4.6 4.6v178.3H54.8v-32c0-2.6-2-4.6-4.6-4.6H22.8c-2.6 0-4.6 2-4.6 4.6V512h182.9v-96c0-72.6 109.7-72.6 109.7 0v96h182.9V292.5c.1-2.6-1.9-4.6-4.5-4.6zm-288.1-4.5c0 2.6-2 4.6-4.6 4.6h-27.4c-2.6 0-4.6-2-4.6-4.6v-64c0-2.6 2-4.6 4.6-4.6h27.4c2.6 0 4.6 2 4.6 4.6v64zm146.4 0c0 2.6-2 4.6-4.6 4.6h-27.4c-2.6 0-4.6-2-4.6-4.6v-64c0-2.6 2-4.6 4.6-4.6h27.4c2.6 0 4.6 2 4.6 4.6v64z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="property-card property-card__title"
    />
    <div
      class="property-card property-card__typeCity"
    >
       - 
    </div>
    <div
      class="property-card"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bath fa-w-16 "
        data-icon="bath"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32,384a95.4,95.4,0,0,0,32,71.09V496a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V480H384v16a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V455.09A95.4,95.4,0,0,0,480,384V336H32ZM496,256H80V69.25a21.26,21.26,0,0,1,36.28-15l19.27,19.26c-13.13,29.88-7.61,59.11,8.62,79.73l-.17.17A16,16,0,0,0,144,176l11.31,11.31a16,16,0,0,0,22.63,0L283.31,81.94a16,16,0,0,0,0-22.63L272,48a16,16,0,0,0-22.62,0l-.17.17c-20.62-16.23-49.83-21.75-79.73-8.62L150.22,20.28A69.25,69.25,0,0,0,32,69.25V256H16A16,16,0,0,0,0,272v16a16,16,0,0,0,16,16H496a16,16,0,0,0,16-16V272A16,16,0,0,0,496,256Z"
          fill="currentColor"
        />
      </svg>
       
    </div>
    <div
      class="property-card"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-bed fa-w-20 "
        data-icon="bed"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M176 256c44.11 0 80-35.89 80-80s-35.89-80-80-80-80 35.89-80 80 35.89 80 80 80zm352-128H304c-8.84 0-16 7.16-16 16v144H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v352c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16v-48h512v48c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V240c0-61.86-50.14-112-112-112z"
          fill="currentColor"
        />
      </svg>
       
    </div>
    <div
      class="property-card property-card__price"
    >
      Â£ 
    </div>
    <div
      class="property-card__email"
    >
      <a
        href="mailto:undefined"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-envelope fa-w-16 "
          data-icon="envelope"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm0 48v40.805c-22.422 18.259-58.168 46.651-134.587 106.49-16.841 13.247-50.201 45.072-73.413 44.701-23.208.375-56.579-31.459-73.413-44.701C106.18 199.465 70.425 171.067 48 152.805V112h416zM48 400V214.398c22.914 18.251 55.409 43.862 104.938 82.646 21.857 17.205 60.134 55.186 103.062 54.955 42.717.231 80.509-37.199 103.053-54.947 49.528-38.783 82.032-64.401 104.947-82.653V400H48z"
            fill="currentColor"
          />
        </svg>
         Email
      </a>
    </div>
  </div>
</DocumentFragment>
`;
